controllers/machine_controller.go

```diff
@@ -598,6 +617,61 @@ func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *MachineReconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	// If there is no tracker, don't watch remote nodes
+	if r.Tracker == nil {
+		return nil
+	}
+
+	if err := r.Tracker.Watch(ctx, remote.WatchInput{
+		Name:         "machine-watchNodes",
+		Cluster:      util.ObjectKey(cluster),
+		Watcher:      r.controller,
+		Kind:         &corev1.Node{},
+		EventHandler: handler.EnqueueRequestsFromMapFunc(r.nodeToMachine),
+	}); err != nil {
```

Comment:
Not necessarily something I think should be blocking for this PR, but I'm wondering if it would make sense to eventually limit the reconciliations triggered by this further to avoid triggering a reconcile when fields that we do not care about are modified. I suspect that would require further changes to the Tracker code to support specifying additional prediicates.

Reply:
The tracker supports predicates - see `WatchInput.Predicates`

Reply:
Ah, indeed, I got mixed up with the inputs to `controllers/external/tracker.go`. My apologies.